### PR TITLE
chore(linter): update golangci.yaml

### PR DESCRIPTION
**Pull request by lasiar**
add new linters

See: 42

### CODE REVIEW

**Review by ccoVeille — CHANGES_REQUESTED** [review]
Dupwords should be enabled, it costs nothing and detects typos

gci should be enabled with the following linter settings

- standard
- default
- localmodule

File: .golangci.yaml [hunk 1]
```diff
@@ -294,6 +306,9 @@ linters:
     # Deprecated. Use govet `fieldalignment`.
     - maligned
 
+    # Reports wrong mirror patterns of bytes/strings usage.
+    - mirror
```
Comment (ccoVeille) [inline .golangci.yaml:310]: Mirror should be enabled

File: .golangci.yaml [hunk 2]
```diff
@@ -404,9 +418,6 @@ linters-settings:
       - name: comment-spacings
         severity: warning
         disabled: false
-        arguments:
```
Comment (ccoVeille) [inline .golangci.yaml:407]: Commenting a random line

Didn't revive support more linter?

**Review by ccoVeille — COMMENTED** [review]
There is no reason to set revive confidence at 0.1 instead of default value 0.8

**Comment by ccoVeille** [comment]
Here is a PR I made for pman

https://github.com/theredditbandit/pman/pull/25

You can see the linters I used. I enabled all revive and disabled only a few, I'm unsure if it was a good idea for maintenance but the results were impressive

**Comment by lasiar** [comment]
> Dupwords should be enabled, it costs nothing and detects typos
> 
> gci should be enabled with the following linter settings
> 
> * standard
> * default
> * localmodule

Done

**Comment by lasiar** [comment]
> Here is a PR I made for pman
> 
> [theredditbandit/pman#25](https://github.com/theredditbandit/pman/pull/25)
> 
> You can see the linters I used. I enabled all revive and disabled only a few, I'm unsure if it was a good idea for maintenance but the results were impressive

I enable all linters from revive
https://github.com/lasiar/canonicalheader/blob/e73e1bb759da822df08218cdbf0db7b4f53248c6/.golangci.yaml#L375-L375

**Comment by lasiar** [comment]
@ccoVeille, I'm sorry for long reply

I'm busy IRL

**Review by ccoVeille — APPROVED** [review]

**Pull request merged**